src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToControlGeneratedAssemblyInfo.cs

```diff
@@ -89,6 +91,175 @@ public void It_respects_opt_outs(string attributeToOptOut)
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject)
+                .WithProjectChanges((path, project) =>
+                {
+                    var ns = project.Root.Name.Namespace;
+
+                    project.Root.Add(
+                        new XElement(ns + "Target",
+                            new XAttribute("Name", "InitializeSourceControlInformation"),
+                            new XElement(ns + "PropertyGroup",
+                                new XElement("SourceRevisionId", ""))));
+
+                    project.Root.Add(
+                        new XElement(ns + "PropertyGroup",
+                            new XElement("SourceControlInformationFeatureSupported", "true"),
+                            new XElement("IncludeSourceRevisionInInformationalVersion", "true")));
+                });
+
+            testAsset.Restore(Log, testProject.Name);
+
+            var command = new GetValuesCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name), "netcoreapp2.0", valueName: "InformationalVersion");
+            command.Execute().Should().Pass();
+
+            command.GetValues().ShouldBeEquivalentTo(new[] { "1.0.0" });
+        }
+
+        [Fact]
+        public void It_does_not_include_source_revision_id_if_disabled()
+        {
+            TestProject testProject = new TestProject()
+            {
+                Name = "ProjectWithSourceRevisionId",
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.0",
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject)
+                .WithProjectChanges((path, project) =>
+                {
+                    var ns = project.Root.Name.Namespace;
+
+                    project.Root.Add(
+                        new XElement(ns + "Target",
+                            new XAttribute("Name", "InitializeSourceControlInformation"),
+                            new XElement(ns + "PropertyGroup",
+                                new XElement("SourceRevisionId", "xyz"))));
+
+                    project.Root.Add(
+                        new XElement(ns + "PropertyGroup",
+                            new XElement("SourceControlInformationFeatureSupported", "true"),
+                            new XElement("IncludeSourceRevisionInInformationalVersion", "false")));
+                });
+
+            testAsset.Restore(Log, testProject.Name);
+
+            var command = new GetValuesCommand(Log, Path.Combine(testAsset.TestRoot, testProject.Name), "netcoreapp2.0", valueName: "InformationalVersion");
+            command.Execute().Should().Pass();
+
+            command.GetValues().ShouldBeEquivalentTo(new[] { "1.0.0" });
+        }
+
+        [Fact]
+        public void It_includes_source_revision_id_if_available__version_without_plus()
```

Comment:
Nit: Looks like there's an extra underscore between "available" and "version" in the name here.

Reply:
On the other hand, perhaps this was intentional to separate the different parts of the test name.

You might consider changing these tests to a Theory.

Reply:
The tests define different XML elements. I'd rather leave them separate.